lifespan/logLifeSpanFactory.go

```diff
@@ -0,0 +1,54 @@
+package lifespan
+
+import (
+	"fmt"
+	"time"
+
+	logger "github.com/ElrondNetwork/elrond-go-logger"
+)
+
+const (
+	epochType    = "epoch"
+	secondType   = "second"
+	megabyteType = "megabyte"
+)
+
+type typeLogLifeSpanFactory struct {
+}
+
+// NewTypeLogLifeSpanFactory creates a new factory for log life spans
+func NewTypeLogLifeSpanFactory() *typeLogLifeSpanFactory {
+	return &typeLogLifeSpanFactory{}
+}
+
+// CreateLogLifeSpanner is a factory method for creating log life spanners
+func (factory *typeLogLifeSpanFactory) CreateLogLifeSpanner(args logger.LogLifeSpanFactoryArgs) (logger.LogLifeSpanner, error) {
+	switch args.LifeSpanType {
+	case epochType:
```

Comment:
why can't we "compose" the log life spanners? I mean, it will be hard to figure out what to write in the config: to create a new log file every epoch or to make-it size capped? We will have to continuously modify the config.toml file. I think it would have been easier to use if we had something like:
create a new file on each epoch change (for example) but if the file exceeds xxx MB, split early.